batch/batch/client.py

```diff
@@ -1,243 +1,131 @@
-import os
+import asyncio
 
-import hailjwt as hj
+from . import aioclient
 
-from . import api
-from .poll_until import poll_until
+
+def async_to_blocking(coro):
+    return asyncio.get_event_loop().run_until_complete(coro)
 
 
 class Job:
     @staticmethod
     def exit_code(job_status):
-        if 'exit_code' not in job_status or job_status['exit_code'] is None:
-            return None
-
-        exit_codes = job_status['exit_code']
-        exit_codes = [exit_codes[task] for task in ['input', 'main', 'output'] if task in exit_codes]
-
-        i = 0
-        while i < len(exit_codes):
-            ec = exit_codes[i]
-            if ec is None:
-                return None
-            if ec > 0:
-                return ec
-            i += 1
-        return 0
+        return aioclient.Job.exit_code(job_status)
+
+    @classmethod
+    def from_async_job(cls, job):
+        j = object.__new__(cls)
+        j._async_job = job
+        return j
 
     def __init__(self, client, id, attributes=None, parent_ids=None, _status=None):
-        if parent_ids is None:
-            parent_ids = []
-        if attributes is None:
-            attributes = {}
+        self._async_job = aioclient.Job(client, id, attributes=attributes,
+                                        parent_ids=parent_ids, _status=_status)
 
-        self.client = client
-        self.id = id
-        self.attributes = attributes
-        self.parent_ids = parent_ids
-        self._status = _status
+    @property
+    def id(self):
+        return self._async_job.id
 
-    def is_complete(self):
-        if self._status:
-            state = self._status['state']
-            if state in ('Complete', 'Cancelled'):
-                return True
-        return False
+    @property
+    def attributes(self):
+        return self._async_job.attributes
 
-    def cached_status(self):
```

Comment:
I don't like this method and I think Cotton probably omitted it on purpose when he wrote aioclient.py. I changed the tests to just print `_status` instead since that is all this is doing and that's the only place it's used.